src/main/java/org/folio/auth/authtokenmodule/storage/ApiTokenStore.java

```diff
@@ -0,0 +1,154 @@
+package org.folio.auth.authtokenmodule.storage;
+
+import org.folio.auth.authtokenmodule.TokenCreator;
+import org.folio.auth.authtokenmodule.tokens.ApiToken;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Tuple;
+
+/**
+ * The API token store provides persistence for API tokens. API tokens need to be
+ * persisted so that they can be revoked.
+ */
+public class ApiTokenStore extends TokenStore {
+  private static final Logger log = LogManager.getLogger(ApiTokenStore.class);
+
+  private static String API_TOKEN_SUFFIX = "api_tokens";
+  private TokenCreator tokenCreator;
+
+  /**
+   * Constructs the store for the given tenant. The tenant can be obtained from
+   * the token when it arrives for validation or when it is created.
+   * @param vertx A reference to the current vertx object.
+   * @param tenant The tenant which is in scope for a given token. The tenant
+   * can be obtained from the token when it arrives for validation or when it is
+   * created.
+   * @param tokenCreator A reference to the the current TokenCreator object
+   * in scope for the main verticle.
+   */
+  public ApiTokenStore(Vertx vertx, String tenant, TokenCreator tokenCreator) {
+    super(vertx, tenant);
+    this.tokenCreator = tokenCreator;
+  }
+
+  /**
+   * Creates the table for this token store if it doesn't yet exist.
+   * @return A failed future should the save operation fail. Otherwise a
+   * succeeded future is returned, even if the table exists.
+   */
+  public Future<Void> createIfNotExists() {
+    // API tokens don't have an owning user. They are associated with a tenant
+    // only. The token itself is persisted since it will need to be viewed by
+    // end-users who have permission to see api tokens.
+    String createTable = "CREATE TABLE IF NOT EXISTS " +
+        tableName(tenant, API_TOKEN_SUFFIX) +
+        "(id UUID PRIMARY key, token TEXT NOT NULL, " +
+        "is_revoked BOOLEAN NOT NULL, issued_at INT8 NOT NULL)";
+
+    log.info("Creating {} tables", TokenStore.class.getName());
+
+    return pool.withConnection(conn -> {
+      return conn.query(createTable).execute().mapEmpty();
+    });
```

Comment:
```suggestion
    return pool.query(createTable).execute().mapEmpty();
```

PgPool has shortcuts for a single query or a single preparedQuery.

Reply:
I think you are right. Will give this a try.